src/openslide-vendor-dicom.c

```diff
@@ -331,6 +349,20 @@ static struct dicom_file *dicom_file_new(const char *filename, GError **err) {
     return NULL;
   }
 
+  const char *syntax = dcm_filehandle_get_transfer_syntax_uid(f->filehandle);
+  uint64_t i;
+  for (i = 0; i < G_N_ELEMENTS(supported_syntax_formats); i++) {
+    if (g_str_equal(syntax, supported_syntax_formats[i].syntax)) {
+      break;
+    }
+  }
+  if (i == G_N_ELEMENTS(supported_syntax_formats)) {
+    g_set_error(err, OPENSLIDE_ERROR, OPENSLIDE_ERROR_FAILED,
+                "Unsupported transfer syntax");
+    return NULL;
+  }
+  f->format = supported_syntax_formats[i].format;
```

Comment:
This could be simplified by moving it to a function with an early return:

```c
static bool get_format(DcmFilehandle *filehandle, enum image_format *format,
                       GError **err) {
  const char *syntax = dcm_filehandle_get_transfer_syntax_uid(filehandle);
  for (uint64_t i = 0; i < G_N_ELEMENTS(supported_syntax_formats); i++) {
    if (g_str_equal(syntax, supported_syntax_formats[i].syntax)) {
      *format = supported_syntax_formats[i].format;
      return true;
    }
  }
  g_set_error(err, OPENSLIDE_ERROR, OPENSLIDE_ERROR_FAILED,
              "Unsupported transfer syntax");
  return false;
}
```

Reply:
OK, done.